### PR TITLE
Improve network manager refresh speed

### PR DESCRIPTION
**Pull request by mansam**
https://bugzilla.redhat.com/show_bug.cgi?id=1512703

### CODE REVIEW

File: app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb [hunk 1]
```diff
@@ -34,10 +34,16 @@ def cloud_networks
     return [] if references(:cloud_networks).blank?
     return @cloud_networks if @cloud_networks.any?
     @cloud_networks = references(:cloud_networks).collect do |network_id|
-      safe_get { network_service.networks.get(network_id) }
+      safe_get { network_service.get_network(network_id).body["network"] }
     end.compact
   end
 
+  def cloud_subnets
```
[the remark this answers is not in the document]
Comment (Ladas) [reply app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb]: hm, I wonder if targeted refresh for subnets works correctly now, the way it is now, they were nested under cloud networks. So for targeted here https://github.com/Ladas/manageiq-providers-openstack/blob/4b784dec74860e678a660002c5d3fca93ace1e33/app/models/manageiq/providers/openstack/inventory/persister/target_collection.rb#L31

we would need to say subnets have ```parent_inventory_collections => :cloud_networks``` and we would need to define ```:targeted_arel``` that returns subnets of all referenced cloud_networks

**Comment by aufi** [comment]
Not sure about @Ladas comment, otherwise looks good to me :+1:

**Comment by Ladas** [comment]
@aufi right, I think we should backport it, then everything will be awesome :-)

**Comment by mansam** [comment]
Since this needs to be backported to Fine (https://github.com/ManageIQ/manageiq/pull/16695) to fix the BZ, it should probably be backported to Gaprindashvili as well.

**Comment by simaishi** [comment]
@mansam If this PR can go to gaprindashvili as is, please add `gaprindashvili/yes` label. Not sure if this will conflict on VCR and you'll need to create a separate PR for gaprindashvili as well...?

**Comment by mansam** [comment]
@simaishi It'll probably conflict so I'll open up a new PR for it

**Comment by simaishi** [comment]
Backported to Gaprindashvili via https://github.com/ManageIQ/manageiq-providers-openstack/pull/251

**Comment by simaishi** [comment]
Backported to Fine via https://github.com/ManageIQ/manageiq/pull/16695

**Pull request merged**